### PR TITLE
Add RGB to Triangle, ITMMesh and the generated OBJ files. Fix #21

### LINKED ISSUE #21: How to add RGB information to mesh?
I would like generate a mesh with x,y,z,r,g,b attributes.
Thanks.

### PR DESCRIPTION
**Pull request by vinjn**
Fix #21

### CODE REVIEW

**Comment by carlren** [comment]
@victorprad could you check @vinjn 's PR and merge it? I'm a little too busy right now :)

File: InfiniTAM/ITMLib/Objects/ITMMesh.h [hunk 1]
```diff
@@ -49,10 +49,11 @@ namespace ITMLib
 				{
 					for (uint i = 0; i < noTotalTriangles; i++)
 					{
-						fprintf(f, "v %f %f %f\n", triangleArray[i].p0.x, triangleArray[i].p0.y, triangleArray[i].p0.z);
-						fprintf(f, "v %f %f %f\n", triangleArray[i].p1.x, triangleArray[i].p1.y, triangleArray[i].p1.z);
-						fprintf(f, "v %f %f %f\n", triangleArray[i].p2.x, triangleArray[i].p2.y, triangleArray[i].p2.z);
-					}
+                        const auto& clr = triangleArray[i].clr;
```
Comment (sgolodetz) [inline InfiniTAM/ITMLib/Objects/ITMMesh.h]: @vinjn: Please can you replace `auto` with `Vector3f`? The former breaks the build on older compilers.

Comment (vinjn) [inline InfiniTAM/ITMLib/Objects/ITMMesh.h]: Sure, what about other changes.

**Comment by connerbrooks** [comment]
@vinjn I made a pull request on your fork which adds rgb info to the CUDA version of the meshing engine and fixes the issue mentioned by @sgolodetz.

**Pull request closed without merging**